src/gf-set-platform

```diff
@@ -41,14 +41,43 @@ coreos_gf_run_mount ro "${tmp_dest}"
 # We just mount the boot partition writable
 coreos_gf remount /boot rw:true
 
+# Look up platform-specific configuration
+rewrite_grub_cmds=
+extra_grub_cmds=
+extra_kargs=
+remove_kargs=
+if [ "$(coreos_gf exists /boot/coreos/platforms.json)" = "true" ]; then
+    coreos_gf download /boot/coreos/platforms.json "${tmpd}"/platforms.json
+    rewrite_grub_cmds=1
+    extra_grub_cmds=$(jq -r ".${platformid}.grub_commands // [] | join(\"\\\\n\")" < "${tmpd}/platforms.json")
+    extra_kargs=$(jq -r ".${platformid}.kernel_arguments // [] | join(\" \")" < "${tmpd}/platforms.json")
+    remove_kargs=$(jq -r ".qemu.kernel_arguments // [] | join(\" \")" < "${tmpd}/platforms.json")
+fi
+
 # Inject PLATFORM label in BLS config (for subsequent config regeneration)
 blscfg_path=$(coreos_gf glob-expand /boot/loader/entries/ostree-*.conf)
 coreos_gf download "${blscfg_path}" "${tmpd}"/bls.conf
 # Remove any platformid currently there
 sed -i -e 's, ignition.platform.id=[a-zA-Z0-9]*,,g' "${tmpd}"/bls.conf
 sed -i -e '/^options / s,$, ignition.platform.id='"${platformid}"',' "${tmpd}"/bls.conf
+if [ -n "$remove_kargs" ]; then
+    # Remove existing qemu-specific kargs
+    sed -i -e '/^options / s@ '"${remove_kargs}"'@@' "${tmpd}"/bls.conf
+fi
```

Comment:
Right.  The problem with such a mark is that it'd be user-visible in the machine's command line for the life of the node.  The advantage to the current approach is that it doesn't have any user-visible side effects.

Reply:
Fair.. Though we *could* push `$ignition_firstboot` (something that already exists that we are adding) to the end of the command line and let that be our mark.

I'm good with what we have here too. Just tossing ideas.

Reply:
Yeah, we could.  It'd give that variable a slightly non-obvious side effect, though.

I'm not too bothered by what we have, though, so I'm inclined to stick with it to avoid another rework.